docs/configuration.md

```diff
@@ -0,0 +1,16 @@
+# fck-nat Configuration
+
+## Configuration file
+
+Upon starting, fck-nat evaluates a configuration file describing how the instance should behave as well as what features
+shall be enabled. To configure fck-nat, ensure a file `/etc/fck-nat.conf` exists with your configuration. fck-nat
+requires the service to be restarted by running `service fck-nat-resart`. In most implementations this configuration is
+passed only once via EC2's user data.
+
+The following describes available options:
+| name                    | description |
+| ----------------------- | ----------- |
+| `eni_id`                | The ID of the Elastic Network Interface to attach to the instance and use as a consistent endpoint to send traffic to fck nat. This is required when using high-availability mode. |
+| `eip_id`                | The ID of an Elastic IP to be attached to the public network interface. This ensures the NAT gateway public traffic is always routed through the same public IP address. |
+| `cwagent_enabled`       | If set, enables Cloudwatch agent and forward instance metrics to Cloudwatch. Requires `cwagent_cfg_param_arn` to be set. |
+| `cwagent_cfg_param_arn` | The ARN of the SSM Parameter holding the Cloudwatch agent configuration and which the agent shall pull from. Requires `cwagent_enabled` to be set. |
```

Comment:
In the docs you refer to the value as `cwagent_cfg_param_arn` but in the script it looks for `cwagent_cfg_param_name` You'll also want to make a note in the docs that for this to work the instance role will need `ssm:GetParameter` permissions as well as any necessary policies for the CloudWatch agent

Reply:
Thanks, nice catch.

As for IAM permissions, I added a section in the section of each feature. I also added for ha-mode regarding ENI and EIP permissions.